### PR TITLE
feat: add tagValueDelimiter parameter

### PR DESCRIPTION
**Pull request by umagnus**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

feat: add tag value delimieter parameter and allow tag values containing commas and equal sign
referring to issue https://github.com/kubernetes-sigs/azuredisk-csi-driver/issues/2323

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/util/util.go [hunk 1]
```diff
@@ -78,27 +77,27 @@ func RoundUpSize(volumeSizeBytes int64, allocationUnitBytes int64) int64 {
 	return roundedUp
 }
 
-// ConvertTagsToMap convert the tags from string to map
+// ConvertTagsToMap convert the tags from string to map, default tagDelimiter is ","
 // the valid tags format is "key1=value1,key2=value2", which could be converted to
 // {"key1": "value1", "key2": "value2"}
-func ConvertTagsToMap(tags string) (map[string]string, error) {
+func ConvertTagsToMap(tags string, tagsDelimiter string) (map[string]string, error) {
```
Comment (andyzhangx) [inline pkg/util/util.go:83]: `tagsDelimiter` could be empty, if it's empty, then it should be `,`

File: pkg/azuredisk/controllerserver.go [hunk 2]
```diff
@@ -924,6 +924,7 @@ func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequ
 	location := d.cloud.Location
 
 	parameters := req.GetParameters()
+	tagValueDelimiter := ","
```
Comment (andyzhangx) [inline pkg/azuredisk/controllerserver.go:927]: just define a var tagValueDelimiter, and then if it's empty, then func `ConvertTagsToMap` would treat as `,`

File: pkg/util/util.go (hunk 1 above)
Comment (andyzhangx) [reply pkg/util/util.go:83]: pls also add unit test for `ConvertTagsToMap`, e.g. `tagsDelimiter` is empty, `,`, `;`

**Comment by andyzhangx** [comment]
@umagnus pls also squash all commits, thx

File: pkg/util/util.go (hunk 1 above)
Comment (umagnus) [reply pkg/util/util.go:83]: done

File: pkg/azuredisk/controllerserver.go (hunk 2 above)
Comment (umagnus) [reply pkg/azuredisk/controllerserver.go]: done

**Review by andyzhangx — APPROVED** [review]

**Pull request merged**